plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveS3InsertOverwrite.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive;
+
+import com.google.common.collect.ImmutableMap;
+import io.trino.plugin.hive.s3.S3HiveQueryRunner;
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.MaterializedResult;
+import io.trino.testing.QueryRunner;
+import io.trino.testing.utils.HiveMinioDataLake;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import static io.trino.testing.MaterializedResult.resultBuilder;
+import static io.trino.testing.sql.TestTable.randomTableSuffix;
+import static java.lang.String.format;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestHiveS3InsertOverwrite
+        extends AbstractTestQueryFramework
+{
+    private static final String HIVE_TEST_SCHEMA = "hive_insert_overwrite";
+
+    private String bucketName;
+    private HiveMinioDataLake dockerizedS3DataLake;
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        this.bucketName = "test-hive-insert-overwrite-" + randomTableSuffix();
+        this.dockerizedS3DataLake = closeAfterClass(
+                new HiveMinioDataLake(
+                        bucketName,
+                        ImmutableMap.of(
+                                "hive_s3_insert_overwrite/hive-core-site.xml", "/etc/hadoop/conf/core-site.xml")));
+        this.dockerizedS3DataLake.start();
+        return S3HiveQueryRunner.create(
+                this.dockerizedS3DataLake.getHiveHadoop().getMappedHiveMetastore(),
+                this.dockerizedS3DataLake.getMinio().getMappedMinioApiHostAndPort(),
+                HiveMinioDataLake.AWS_ACCESS_KEY,
+                HiveMinioDataLake.AWS_SECRET_KEY,
+                ImmutableMap.<String, String>builder()
+                        .put("hive.s3.path-style-access", "true")
+                        .put("hive.immutable-partitions", "false")
+                        .put("hive.insert-existing-partitions-behavior", "OVERWRITE")
```

Comment:
Does OVERWRITE require mutable partitions? If so, make sure we document this in this PR if not already done.

Reply:
No I've overlooked that. Will remove this property from test.